### PR TITLE
#162949032 Setup react router version 4 (React-router-dom)

### PR DESCRIPTION
**Pull request by sharkdevs**
#### What does this PR do?

Enables routing with react router version 4 

#### Description of Task to be completed?

- [x] Create a components folder
- [x] Create a routes folder
- [x] Add home, log in and error components and views
- [x] route the pages

#### How should this be manually tested?

1. Clone this repository on your local machine using 

     `git clone https://github.com/andela/ah-robotics-frontend.git`
2. Change your working directory to the project's directory using 
     
     `cd ah-robotics-frontend`
3. Change to the specific branch `ch-react-router-v4-162949032` using 

     `git checkout ch-react-router-v4-162949032`

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm start` command to start the application.

6. The application should start successfully displaying a page as shown below.

screen shot 2019-02-07 at 18 04 43

7. On the top navigation bar Click on the `Log In` link to go to the Login page below:

screen shot 2019-02-07 at 18 04 22

8. The rest of the links will display the image below because they are not available:

screen shot 2019-02-07 at 18 09 31

#### What are the relevant pivotal tracker stories?

[#162949032](https://www.pivotaltracker.com/story/show/162949032)

[Finishes #162949032]

### CODE REVIEW

**Comment by Teatoller** [comment]
Hello @sharkdevs ,

I am testing out 
>#162949032 Setup react router version 4 (React-router-dom) #8

However at step 4 when I try to run the ```npm install``` I am encountering the error on my terminal as captured on the screenshot.
screenshot 2019-02-10 at 20 45 39

Could you check into it as it not possible to progress on to step ```5```.

Regards
Steven Ennis

**Comment by sharkdevs** [comment]
> Hello @sharkdevs ,
> 
> I am testing out
> 
> > #162949032 Setup react router version 4 (React-router-dom) #8
> 
> However at step 4 when I try to run the `npm install` I am encountering the error on my terminal as captured on the screenshot.
> screenshot 2019-02-10 at 20 45 39
> 
> Could you check into it as it not possible to progress on to step `5`.
> 
> Regards
> Steven Ennis

@Teatoller I have fixed the issues, it is now fine. You can proceed.

**Comment by jomasim** [comment]
Check your commit messages and ensure to change them to lowercase according to the [engineering playbook](https://github.com/andela/engineering-playbook/tree/master/5.%20Developing/Conventions)

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+/* eslint-disable no-unused-vars */
```
Comment (Collins33) [inline src/App.js]: What is this for?

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
-);
+import Home from './views/HomeView';
+import Login from './views/LoginView';
+import Error from './views/ErrorView';
+import Navigation from './views/NavigationView';
+
+
+class App extends Component {
+	render() {
+		return (
```
Comment (Collins33) [inline src/App.js]: Why is the indentation like this?

File: src/App.test.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
+/* eslint-disable no-undef */
```
Comment (Collins33) [inline src/App.test.js]: What are this for?

File: src/App.test.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 import { shallow } from 'enzyme';
 import App from './App';
 
 describe('RootComponent', () => {
-  it('app should render without crashing', () => {
-    const wrapper = shallow(<App />);
-    expect(wrapper.length).toBe(1);
-  });
+	it('app should render without crashing', () => {
+		const wrapper = shallow(<App />);
```
Comment (Collins33) [inline src/App.test.js]: Why is the indentation like this?

File: src/components/Error.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// eslint-disable-next-line no-unused-vars
+import React from 'react';
+
+const Error = () =>{
+	return(
+		<div>
```
Comment (Collins33) [inline src/components/Error.js]: Same here

File: src/components/Home.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-unused-vars
+import React from 'react';
+import logo from '../logo.svg';
+import '../App.css';
+
```
Comment (Collins33) [inline src/components/Home.js:4]: Too much whitespace

File: src/components/Login.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-unused-vars */
+import React from 'react';
+import { Button, Form, Grid, Header, Image, Message, Segment } from 'semantic-ui-react';
+
+// Create a login component view
+// This code is a modification of the sample login page in Semantic UI
+const LoginView = () =>
+	(
+		<div className='login-form'>
+			<style>
```
Comment (Collins33) [inline src/components/Login.js]: Same here

File: src/components/Navigation.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+
+// Create a navigation Component View
+const NavigationView = () => (
+	<div>
+
+		<Menu text>
+			<NavLink to="/">
+				<Menu.Item name="Home" />
+			</NavLink>
```
Comment (Collins33) [inline src/components/Navigation.js]: Same here

**Review by Collins33 — CHANGES_REQUESTED** [review]
The indentation on most files seems off. Fix them

File: src/App.js (hunk 1 above)
Comment (sharkdevs) [reply src/App.js]: eslint highlights variables that do not appear to be used and thus affect codacy. For example, React import. Eslint provides those inline comments to fix the issues.

File: src/App.js (hunk 2 above)
Comment (sharkdevs) [reply src/App.js]: it is set by eslint, the indentation is set to 1 tabspace. But I have to reformat it to 2 white spaces.

**Review by waracci — COMMENTED** [review]
Looks good, just ensure you fix the merge conflicts.

**Review by johnwayodi — COMMENTED** [review]
@sharkdevs Good work. I managed to test the feature after the requested changes and it is working well.

**Review by Collins33 — APPROVED** [review]

**Pull request merged**